src/test/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/ReferenceDataControllerTest.java

```diff
@@ -0,0 +1,122 @@
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.SessionType;
+import uk.gov.hmcts.reform.sandl.snlevents.model.response.CaseTypeWithHearingTypesResponse;
+import uk.gov.hmcts.reform.sandl.snlevents.model.response.SimpleDictionaryData;
+import uk.gov.hmcts.reform.sandl.snlevents.security.S2SAuthenticationService;
+import uk.gov.hmcts.reform.sandl.snlevents.service.ReferenceDataService;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(ReferenceDataController.class)
+@Import(TestConfiguration.class)
+public class ReferenceDataControllerTest {
+    public static final String URL_ROOM_TYPES = "/reference/room-types";
+    public static final String URL_SESSION_TYPES = "/reference/session-types";
+    public static final String URL_HEARING_TYPES = "/reference/hearing-types";
+    public static final String URL_CASE_TYPES = "/reference/case-types";
+
+    private static final String DEFAULT_CODE = "sample-code";
+    private static final String DEFAULT_DESCRIPTION = "sample description";
+
+    @Autowired
+    private EventsMockMvc mvc;
+
+    @MockBean
+    private ReferenceDataService testService;
+    @MockBean
+    @SuppressWarnings("PMD.UnusedPrivateField")
+    private S2SAuthenticationService s2SAuthenticationService;
+
+    @Before
+    public void setupMock() {
+        when(s2SAuthenticationService.validateToken(any())).thenReturn(true);
+    }
+
+    @Test
+    public void fetchAllRoomTypes_returnsRoomTypesFromService() throws Exception {
+        val mockedValues = createRoomTypes();
+        when(testService.getRoomTypes()).thenReturn(mockedValues);
+
+        val response = mvc.getAndMapResponse(URL_ROOM_TYPES, new TypeReference<List<SimpleDictionaryData>>() {
+        });
+        assertThat(response).isEqualTo(mockedValues);
+    }
+
+    @Test
+    public void fetchAllSessionTypes_returnsSessionTypesFromService() throws Exception {
+        val mockedValues = createSessionTypes();
+        when(testService.getSessionTypes()).thenReturn(mockedValues);
+
+        val response = mvc.getAndMapResponse(URL_SESSION_TYPES, new TypeReference<List<SimpleDictionaryData>>() {
+        });
+        assertThat(response).isEqualTo(mockedValues);
+    }
+
+    @Test
+    public void fetchAllHearingTypes_returnsHearingTypesFromService() throws Exception {
+        val mockedValues = createHearingTypes();
+        when(testService.getHearingTypes()).thenReturn(mockedValues);
+
+        val response = mvc.getAndMapResponse(URL_HEARING_TYPES, new TypeReference<List<SimpleDictionaryData>>() {
+        });
+        assertThat(response).isEqualTo(mockedValues);
+    }
+
+    @Test
+    public void fetchAllCaseTypes_returnsCaseTypesFromService() throws Exception {
```

Comment:
it would be good to have unit test that cover connection case types with hearing types

Reply:
this already tests such case

Reply:
quick one question, what if the controller fails to fetch any reference data , do we need to do handle no response received for that call ?